### PR TITLE
Minor cleanups in ESPClass

### PR DESCRIPTION
**Pull request by devyte**
Related to #6599

### CODE REVIEW

File: cores/esp8266/Esp.cpp [hunk 1]
```diff
@@ -579,24 +579,25 @@ bool EspClass::flashEraseSector(uint32_t sector) {
...
-    int rc = 0;
-    uint32_t* ptr = data;
 
     if (flash_write_puya_buf == nullptr) {
         flash_write_puya_buf = (uint32_t*) malloc(PUYA_BUFFER_SIZE);
         // No need to ever free this, since the flash chip will never change at runtime.
         if (flash_write_puya_buf == nullptr) {
             // Memory could not be allocated.
-            return 1; // SPI_FLASH_RESULT_ERR
+            return SPI_FLASH_RESULT_ERR; // SPI_FLASH_RESULT_ERR
```
Comment (dirkmueller) [inline cores/esp8266/Esp.cpp]: can you remove the now redundant comment?

File: cores/esp8266/Esp.cpp [hunk 2]
```diff
@@ -579,24 +579,25 @@ bool EspClass::flashEraseSector(uint32_t sector) {
 #if PUYA_SUPPORT
 static int spi_flash_write_puya(uint32_t offset, uint32_t *data, size_t size) {
     if (data == nullptr) {
-      return 1; // SPI_FLASH_RESULT_ERR
+      return SPI_FLASH_RESULT_ERR; // SPI_FLASH_RESULT_ERR
```
Comment (dirkmueller) [inline cores/esp8266/Esp.cpp]: can you remove the now redundant comment?

File: cores/esp8266/Esp.cpp (hunk 1 above)
Comment (devyte) [reply cores/esp8266/Esp.cpp]: Done

File: cores/esp8266/Esp.cpp (hunk 2 above)
Comment (devyte) [reply cores/esp8266/Esp.cpp]: Done

**Pull request merged**